src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -871,6 +873,19 @@ class Namer { typer: Typer =>
       sym.info = NoCompleter
       sym.info = checkNonCyclic(sym, unsafeInfo, reportErrors = true)
     }
+
+    // Here we pay the price for the cavalier setting info to TypeBounds.empty above.
```

Comment:
The TypeBounds.empty thing is tricky and I don't really understand why it's done and how it works, I think it'd be great to explain it in more details, for example there's a comment above that says:
```scala
// The scheme critically relies on an implementation detail of isRef, which
// inspects a TypeRef's info, instead of simply dealiasing alias types.
```
But why would this scheme not work if isRef was implemented differently? (I don't even see how you could implement it differently, how can you know if an alias TypeRef refers to a particular symbol without looking at its info?)